src/Workspaces/Core/Portable/Utilities/Operations/IOperationContext.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.Utilities
+{
+    /// <summary>
+    /// Represents a context of executing potentially long running operation with possible wait indication providing
+    /// cancellability, progress and information to the host.
+    /// </summary>
+    public interface IOperationContext : IDisposable
+    {
+        /// <summary>
+        /// Cancellation token that allows user to cancel the operation unless the operation is not cancellable.
+        /// </summary>
+        CancellationToken CancellationToken { get; }
+
+        /// <summary>
+        /// Gets user readable operation description, composed of initial context description and descriptions of all
+        /// currently added scopes.
+        /// </summary>
+        string Description { get; }
+
+        /// <summary>
+        /// Gets current list of <see cref="IOperationScope"/>s in this context.
+        /// </summary>
+        IEnumerable<IOperationScope> Scopes { get; }
```

Comment:
>IEnumerable [](start = 8, length = 11)

ImmutableArray?

Reply:
From an impl perspective, it's not (it's an ImmutableList).  This matches the VS api here.  We coudl always 'reify' the listif you did call this though.  so i'm not beholden to this being an IEnumerable.

Reply:
I don't think we necessarily need to match VS APIs.

---
In reply to: [551563773](https://github.com/dotnet/roslyn/pull/50211#discussion_r551563773) [](ancestors = 551563773)